### ISSUE TITLE
Learning loop doesn't create converging logits

### ISSUE THREAD

**Issue by ariasanovsky** [ariasanovsky/azdopt#40]
The cross entropy error hovers around `5.84` in example `07`. To address this, we create method to sample the space better between batches. At the end of every `K`-th epoch:

- [x] sort visited states by cost, e.g., a `Vec<(&S, &P, f32)>`
- [x] split into `BATCH` equally-sized chunks
- [x] root the next search tree at `P @ S`
- avoid terminal nodes
- consider resetting the prohibited actions of some
- [x] implement in search

**Comment by ariasanovsky** [comment]
# New requirement

- [x] refactor plotting to show argmin
- [x] refactor plotting to show losses
- [x] refactor plotting to show weights and biases
- probably we remove all the old logs
  - we don't want our logger to also be our root selection algorithm
  - since batch size is huge, only track the argmin from each episode
  - if it decreases the running argmin, push it onto the queue and update the serialized data

# Tensorboard crate

- This will save us having to marshall data for numerical plots to `python`
- We can still plot graphs and trees with `networkx`
- TB costs us a thread since it hosts a site at runtime, but we can disable and rerun later
